code/datums/components/barber.dm

```diff
@@ -0,0 +1,605 @@
+#define HAIRCUT 1
+#define SHAVE 2
+#define BARBERY_FAILURE 4	// if barbering is not successful and does not display a message
+#define BARBERY_SUCCESSFUL 1 // if barbering is successful, don't attack em
+#define BARBERY_RESOLVABLE 2 // if barbering is not successful, but gives a message
+
+/datum/component/toggle_tool_use
+/datum/component/toggle_tool_use/Initialize()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, list(COMSIG_ITEM_DROPPED), .proc/on_drop)
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_SELF), .proc/on_attackself)
+
+	// this proc is supposed to make certain tools less accidentally deadly for inexperienced players to use
+	// when force_use_as_tool is set, all intents will try to do their tool-thing, and if it can't, return a message saying they're using it wrong
+	// if not set, help intent will still attempt tool, but you'll shank them if it doesn't work out
+/datum/component/toggle_tool_use/proc/on_drop(var/obj/item/thing)
+	toggle_force_use_as_tool(thing)
+
+/datum/component/toggle_tool_use/proc/on_attackself(var/obj/item/thing, mob/user)
+	toggle_force_use_as_tool(thing, user)
+
+/datum/component/toggle_tool_use/proc/toggle_force_use_as_tool(var/obj/item/thing, mob/user)
+	if(!user)
+		thing.force_use_as_tool = 0
+		thing.dir = SOUTH
+		return
+	thing.force_use_as_tool = !thing.force_use_as_tool
+	if (thing.force_use_as_tool)
+		thing.dir = WEST
+	else
+		thing.dir = SOUTH
```

Comment:
This dir changeing definitely deserves a comment explaining it because it's not clear at all. I assume you are toggling different "icon states" based on the tool mode? Doing it through dir seems really janky but I guess it can stay even if I don't like it.

An alternative cleaner way of doing this would be for the component to take arguments in `Initialize` that say what the icon state is supposed to be in tool mode and what it's supposed to be in non-tool mode and then storing and switching between those.

Reply:
I used dir as a way to both change what it looks like in hand, and to let the tools look different when thrown. After all, changing the way you hold a pair of scissors doesn't really change how the thing looks, just what direction it's pointing.

I suppose it would be kind of weird for tools that'd need to change icon_states when toggled. But, those don't exist _yet_, and using the component to change icon_state would need anything that does change icon_state to use a specific naming format (scissors and scissors-toolmode), and/or a defined icon_state_toolmode var. And might cause problems if those arent defined.

I could define and explain the dirs in init, though.

Reply:
Probably just add a comment for now.